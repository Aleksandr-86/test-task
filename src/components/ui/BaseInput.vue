<script setup lang="ts">
import { ref } from '@vue/reactivity'

defineProps<{
  label: string
  type: string
  id: string
  autocomplete: string
  iconSrc?: string // Наличие иконки слева
  showPassword?: boolean // Наличие иконки показа скрытого пароля
}>()

const inputContentRef = ref('')
const typeRef = ref('password')

const changeInputType = () => {
  if (typeRef.value === 'password') {
    typeRef.value = 'text'
  } else {
    typeRef.value = 'password'
  }
}
</script>

<template>
  <div class="flex flex-col space-y-2">
    <div class="h-6">
      <transition
        enter-from-class="opacity-0"
        leave-to-class="opacity-0"
        enter-active-class="transition linear duration-700"
        leave-active-class="transition linear duration-700">
        <label v-if="inputContentRef" class="text-lg font-bold" :for="id">
          {{ label }}
        </label>
      </transition>
    </div>

    <div class="relative">
      <input
        class="h-10 w-full rounded text-2xl shadow-[inset_1px_1px_2px_rgba(0,0,0,0.5)]"
        :class="[
          { 'px-5': !iconSrc },
          { 'pe-5 ps-10': iconSrc && !showPassword },
          { 'px-10 ': iconSrc && showPassword }
        ]"
        :type="type === 'password' ? typeRef : type"
        :name="id"
        :id="id"
        :placeholder="label"
        :autocomplete="autocomplete"
        v-model="inputContentRef" />

      <!-- Иконка слева -->
      <div v-if="iconSrc" class="absolute bottom-0 left-0 top-0 flex p-2">
        <img :src="iconSrc" />
      </div>

      <!-- Иконка справа -->
      <transition
        enter-from-class="opacity-0"
        leave-to-class="opacity-0"
        enter-active-class="transition linear duration-700"
        leave-active-class="transition linear duration-700">
        <img
          v-if="showPassword && inputContentRef"
          @click="changeInputType"
          class="absolute right-2 top-2 h-6 w-6 cursor-pointer"
          :src="
            typeRef === 'text' ? 'icons/eye-closed.svg' : 'icons/eye-opened.svg'
          " />
      </transition>
    </div>
  </div>
</template>
