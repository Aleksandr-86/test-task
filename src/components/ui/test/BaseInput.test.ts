import { render } from '@testing-library/vue'
import BaseInput from '@/components/ui/BaseInput.vue'

describe('Компонент BaseInput', () => {
  it('структура соответствует снимку', () => {
    const wrapper = render(BaseInput, {
      props: {
        label: 'Поле',
        type: 'text',
        id: 'field',
        autocomplete: 'password',
        iconSrc: 'icons/key',
        showPassword: true
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
