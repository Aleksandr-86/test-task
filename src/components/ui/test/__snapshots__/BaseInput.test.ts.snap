// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Компонент BaseInput > структура соответствует снимку 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col space-y-2"
      >
        <div
          class="h-6"
        >
          <transition-stub
            appear="false"
            css="true"
            enteractiveclass="transition linear duration-700"
            enterfromclass="opacity-0"
            leaveactiveclass="transition linear duration-700"
            leavetoclass="opacity-0"
            persisted="false"
          >
            <!--v-if-->
          </transition-stub>
        </div>
        <div
          class="relative"
        >
          <input
            autocomplete="password"
            class="outline-solid h-10 w-full rounded text-2xl shadow-[inset_1px_1px_2px_rgba(0,0,0,0.5)] outline-blue-700 dark:bg-gray-200 px-10"
            id="field"
            name="field"
            placeholder="Поле"
            type="text"
          />
          <!-- Иконка слева -->
          <div
            class="absolute bottom-0 left-0 top-0 flex p-2"
          >
            <img
              src="icons/key"
            />
          </div>
          <!-- Иконка справа -->
          <transition-stub
            appear="false"
            css="true"
            enteractiveclass="transition linear duration-700"
            enterfromclass="opacity-0"
            leaveactiveclass="transition linear duration-700"
            leavetoclass="opacity-0"
            persisted="false"
          >
            <!--v-if-->
          </transition-stub>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col space-y-2"
    >
      <div
        class="h-6"
      >
        <transition-stub
          appear="false"
          css="true"
          enteractiveclass="transition linear duration-700"
          enterfromclass="opacity-0"
          leaveactiveclass="transition linear duration-700"
          leavetoclass="opacity-0"
          persisted="false"
        >
          <!--v-if-->
        </transition-stub>
      </div>
      <div
        class="relative"
      >
        <input
          autocomplete="password"
          class="outline-solid h-10 w-full rounded text-2xl shadow-[inset_1px_1px_2px_rgba(0,0,0,0.5)] outline-blue-700 dark:bg-gray-200 px-10"
          id="field"
          name="field"
          placeholder="Поле"
          type="text"
        />
        <!-- Иконка слева -->
        <div
          class="absolute bottom-0 left-0 top-0 flex p-2"
        >
          <img
            src="icons/key"
          />
        </div>
        <!-- Иконка справа -->
        <transition-stub
          appear="false"
          css="true"
          enteractiveclass="transition linear duration-700"
          enterfromclass="opacity-0"
          leaveactiveclass="transition linear duration-700"
          leavetoclass="opacity-0"
          persisted="false"
        >
          <!--v-if-->
        </transition-stub>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
